tests/functional/test_basic.py

```diff
@@ -380,6 +380,18 @@ def test_set_icon(pyi_builder, data_dir):
     pyi_builder.test_source("print('Hello Python!')", pyi_args=args)
 
 
+def test_osx_app_console_option(pyi_builder, data_dir):
+    if is_darwin:
+        # On OS X icon is applied only for windowed mode.
+        args = ['--windowed', '--osx-app-console']
+    else:
+        pytest.skip('option --icon works only on Windows and Mac OS X')
+    pyi_builder.test_source("""
+        print('Hello Python!')
+        input()
+    """, pyi_args=args)
```

Comment:
This looks suspicious, as it will wait for input forever. And `pyi_builder.test_source()` probably runs the `onedist`/`onefile` version of the build (instead of app bundle, which is created in addition to one of those builds), so the test will be stuck until it times out. Even if the bundle was ran instead, it would remain open forever, and we'd have no way of knowing its status (because opening a bundle returns immediately)...

So the test should be modeled after tests from [`test_apple_events.py`](https://github.com/pyinstaller/pyinstaller/blob/develop/tests/functional/test_apple_events.py): freeze an app bundle, whose code checks if console is available (e.g., by checking `sys.stdin.isatty()`), and writes the console status into a file (whose path can be injected into the script source via format(), or perhaps via environment variables); the main test then calls the bundle via `subprocess` and `open` call, then waits for some time (10 seconds?), and checks if file exists and if its content indicates that the console was available to the bundle.

This way, we'll know that the app bundle ran successfully and that it had console.

And the test app bundle should have a space in its name, to explicitly test for that.